### PR TITLE
feat: add cycle sort algorithm cycle_sort2.cpp

### PR DESCRIPTION
**Pull request by VasuSehgal123**
#### Description of Change
Cycle sort is based on the idea that the array to be sorted can be divided into cycles, which can individually be rotated to give a sorted result. It is an in-place, unstable sorting algorithm, a comparison sort that is theoretically optimal in terms of the total number of memory writes to the original array. 
Example 1:
Input : { 1, 8, 3, 9, 10, 10, 2, 4 }
Output : { 1, 2, 3, 4, 8, 9, 10, 10 }

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1332"></a>

### CODE REVIEW

**Comment by Panquesito7** [comment]
Duplicate of #1252.

**Comment by Panquesito7** [comment]
You may want to change the filename to `cycle_sort2.cpp`, as the PR #1252 implements the same algorithm. 🙂

**Comment by VasuSehgal123** [comment]
@Panquesito7 Changed  filename from `cycle_sort.cpp` to `cycle_sort2.cpp`

File: sorting/cycle_sort2.cpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+ * or written one time to its correct position.
+ *
+ * The array is divided into cycles. We begin with the cycle containing the first element. 
+ * Find the correct position of first element and place it at its correct position, say j. 
+ * Now we consider the old value of arr[j] and find its correct position. 
+ * This process is continued until all the elements of the current cycle are placed at their correct position,
+ * i.e.  until the starting point of the cycle is reached.
+ * 
+ * @author [Vasu Sehgal](https://www.github.com/VasuSehgal123)
+ * @see more on [Cycle sort] (https://en.wikipedia.org/wiki/Cycle_sort)
```
Comment (Panquesito7) [inline sorting/cycle_sort2.cpp:19]:
```suggestion
 * @see more on [Cycle sort](https://en.wikipedia.org/wiki/Cycle_sort)
```

File: sorting/cycle_sort2.cpp [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+	}  // namespace cycle_sort
+	
+}  // namespace sorting
+
+/**
+ * @brief Test implementations
+ * @returns void
+ */
+ void test() {
```
Comment (Panquesito7) [inline sorting/cycle_sort2.cpp:114]:
```suggestion
 static void test() {
```

File: sorting/cycle_sort2.cpp [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+  
+int main() 
+{  
+   test();
+    return 0; 
+}
```
Comment (Panquesito7) [inline sorting/cycle_sort2.cpp:146]:
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/VasuSehgal123/C-Plus-Plus/actions

File: sorting/cycle_sort2.cpp [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        std::cout << arr2[i] << "  ";
+     std::cout<<"\n";
+
+
+}
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+
```
Comment (Panquesito7) [inline sorting/cycle_sort2.cpp:141]:
```suggestion
```

File: sorting/cycle_sort2.cpp [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+}
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main()
+{
+   test();
+    return 0;
```
Comment (Panquesito7) [inline sorting/cycle_sort2.cpp:144]:
```suggestion
   return 0;
```

File: sorting/cycle_sort2.cpp [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+
+}
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main()
+{
+   test();
```
Comment (Panquesito7) [inline sorting/cycle_sort2.cpp:143]:
```suggestion
   test();  // execute the tests
```

**Review by Panquesito7 — APPROVED** [review]
Nice work; I like how you've made the code. 😄 👍
Code and documentation is pretty good and refined; LGTM.

**Comment by VasuSehgal123** [comment]
Hi @Panquesito7! Directory.md is showing merge conflict after #1252 got merged. Can you do something about it?

**Comment by Panquesito7** [comment]
> Can you do something about it?

Merge conflict should be resolved now. 🙂

**Pull request merged**